XenAdmin/Actions/OVFActions/ApplianceAction.cs

```diff
@@ -97,5 +100,21 @@ protected override void CancelRelatedTask()
 			if (m_transportAction != null)
 				m_transportAction.Cancel = true;
 		}
+
+	    protected void InitialiseTicker()
+	    {
+	        System.Threading.Tasks.Task.Run(() => TickUntilCompletion());
+	    }
+
+	    private void TickUntilCompletion()
+	    {
+	        while (!IsCompleted)

```

Comment:
Could we add a null check here? I'm a bit paranoid the action might be nullified on a different thread and this one can't notice.